analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -411,11 +411,17 @@ public static bool IsComment(this SyntaxTrivia trivia)
         public static ImmutableArray<ExpressionSyntax> ArgumentValuesForParameter(SemanticModel semanticModel, ArgumentListSyntax argumentList, string parameterName)
         {
             var methodParameterLookup = new CSharpMethodParameterLookup(argumentList, semanticModel);
-            if (methodParameterLookup.TryGetSyntax(parameterName, out var argumentSyntax))
-            {
-                return argumentSyntax.Select(x => x.Expression).ToImmutableArray();
-            }
-            return ImmutableArray<ExpressionSyntax>.Empty;
+            return methodParameterLookup.TryGetSyntax(parameterName, out var argumentSyntax)
+                ? argumentSyntax.Select(x => x.Expression).ToImmutableArray()
+                : ImmutableArray<ExpressionSyntax>.Empty;
         }
+
+        // (arg, b) = something
+        public static bool IsAssignmentToTuple(this ArgumentSyntax argument) =>
+            argument.Parent is { } tupleExpression
+            && TupleExpressionSyntaxWrapper.IsInstance(tupleExpression)
+            && tupleExpression.Parent is AssignmentExpressionSyntax assignment
+            && assignment.Left == tupleExpression;
```

Comment:
I find it easier to follow if you write
```
&& tupleExpression == assignment.Left;
```
because we read left to rigth (and at this point, we're focused on the tuple)

Reply:
Last 2 lines
```
            && tupleExpression.Parent is AssignmentExpressionSyntax assignment
            && assignment.Left == tupleExpression;
```
are inspecting the `assignment`. So to me it's like "it's assignment and the assignment matches the constraint". It also follow the logic of going deeper in the structure.